File: swift/Sources/sdnx/Types/ArraySchema.swift
import Foundation

public struct ArraySchema: SchemaValue {
    public let type: String = "array"
    public var description: String?
    public var validators: [String: ValidatorInfo]?
    public let inner: SchemaValue
    
    public init(inner: SchemaValue, description: String? = nil, validators: [String: ValidatorInfo]? = nil) {
        self.inner = inner
        self.description = description
        self.validators = validators
    }
}

File: dotnet/Sdnx.Core/Types/ArraySchema.cs
namespace Sdnx.Core
{
    public record ArraySchema : SchemaValue
    {
        public required string Type { get; init; } = "array";
        public required SchemaValue Inner { get; init; }
        public string? Description { get; init; }
        public Dictionary<string, Validator>? Validators { get; init; }
    }
}
